kaldb/src/main/java/com/slack/kaldb/chunkManager/CachingChunkManager.java

```diff
@@ -49,6 +53,14 @@ public CachingChunkManager(
     this.s3Bucket = s3Bucket;
     this.dataDirectoryPrefix = dataDirectoryPrefix;
     this.slotCountPerInstance = slotCountPerInstance;
+
+    // todo - consider making the thread count a config option; this would allow for more
+    //  fine-grained tuning, but we might not need to expose this to the user if we can set sensible
+    //  defaults
+    this.executorService =
+        Executors.newFixedThreadPool(
+            Runtime.getRuntime().availableProcessors() >= 4 ? 2 : 1,
+            new ThreadFactoryBuilder().setNameFormat("caching-chunk-manager-%d").build());
```

Comment:
It would be good to include an uncaught exception handler here, too.

Reply:
Done.